app/javascript/controllers/qr_code_scanner_controller.js

```diff
@@ -0,0 +1,37 @@
+import { Controller } from "@hotwired/stimulus";
+import QrScanner from "qr-scanner";
+
+export default class extends Controller {
+  connect() {
+    this.isScanning = false;
+    this.qrScanner = null;
+    document.querySelector("#scanner").addEventListener("click", () => this.scan());
+  }
+
+  scan() {
+    document.querySelector(".qr-code-scanner").classList.remove("d-none");
+    this.qrScanner = new QrScanner(
+        document.querySelector("video"),
+        result => this.handleScan(result),
+    );
+    this.isScanning = true;
+    this.qrScanner.start();
+    document.querySelector("#close").addEventListener("click", () => this.stopScan());
+  }
+
+  stopScan() {
+    this.isScanning = false;
+    this.qrScanner.stop();
+    document.querySelector(".qr-code-scanner").classList.add("d-none");
+  }
+
+  handleScan(result) {
+    if (result.includes("/i/r") && this.isScanning) {
```

Comment:
Je suis pas sûr d'avoir compris à quoi sert `this.isScanning`, ça veut dire qu'on peut appeler cette fonction alors qu'on est pas en train de scanner?

Reply:
En fait c'est pour debounce le truc, car la fonction de scan hit pas mal de fois à la seconde et je voulais éviter d'appeler cette fonction plus d'une fois